src/Journal.cpp

```diff
@@ -83,29 +152,23 @@ void Journal::recordUndoAction (
   const std::string &before,
   const std::string &after)
 {
-  if (_currentTransaction != nullptr)
+  if (enabled ())
   {
-    _currentTransaction->addUndoAction (type, before, after);
+    if (_currentTransaction != nullptr)
+    {
+      _currentTransaction->addUndoAction (type, before, after);
+    }
   }
 }
 
 ////////////////////////////////////////////////////////////////////////////////
+// Run popLastTransaction even if the journal is disabled. Otherwise, a user
```

Comment:
Again, good point. I was thinking that it would be good to allow the user to undo the config change, but your next comment shows by that is problematic.